### ISSUE TITLE
[Suggestion] [Resource Packs] Block LOD

### ISSUE THREAD

**Issue by Hax4Ever** [sp614x/optifine#2221]
_tl;dr-The title really says it all I can't say it much better than that._

I'm a resource pack artist, and after months of fiddling with what I have made, I came to the conclusion that the leaves in my pack were causing a bit of a hit on the performance. I, however, don't want to sacrifice too much detail just to keep the game running above 60fps, and thought a great middle ground would be some sort of LOD system that would switch models out based on what is near the player and what isn't near the player. One possible method could be just having a near and far plane and after 4 chunks or so it'd use the low detail models instead of the high detail ones, or a pack creator could define multiple levels for distance.

**Comment by sp614x** [comment]
This quite complicated to be implemented. 
Side effects which can't be avoided:
- the chunk loading would be 2x slower
- the 1.13 chunk loading is already 2x slower than 1.12 (waterlogging), so it wiould become even worse
- the GPU upload bandwidth would increase 2x
- chunk loading lag spikes would become significant

**Comment by sp614x** [comment]
If the chunks are split in parts (non-LOD blocks, LOD blocks) it may be possible to reduce the side effects. 
This may conflict with some mods, the CPU rendering load would increase and LOD couldn't be used for translucent blocks as they need a specific sorting order and can't be split.

**Comment by sp614x** [comment]
Adding LOD rendering layers may be used to switch between normal/LOD models.

**Comment by Hax4Ever** [comment]
> 
> The rendering itself may be slowed down a bit as it has to go over more layers.
Though it'd be a good trade off in the instance you have a very complicated recurring model (read : high detail leaves)
